src/markdown_exec/__init__.py

```diff
@@ -124,7 +130,10 @@ def formatter(
         HTML contents.
     """
     fmt = formatters.get(language, lambda source, **kwargs: source)
-    return fmt(code=source, md=md, **options)  # type: ignore[operator]
+    try:
+        return fmt(code=source, md=md, **options)  # type: ignore[operator]
+    except Exception as e:
+        raise ExecutionError(traceback.format_exc()) from e
```

Comment:
issue: Since we're returning a Python traceback, this should be formatted as a Python code block for proper display in the rendered docs.